mri/tests/test_sensitivity_extraction.py

```diff
@@ -84,6 +84,32 @@ def test_extract_k_space_center_2D(self):
             center_Img.reshape(data_thresholded.shape),
             data_thresholded)
 
+    def test_extract_k_space_center_2D_fft(self):
+        """ Ensure that the extracted k-space center is right
+        for cartesian case"""
+        mask = np.random.randint(0, 2, (self.N, self.N))
+        samples = convert_mask_to_locations(mask)
+        Img = (np.random.randn(self.num_channel, self.N, self.N) +
+               1j * np.random.randn(self.num_channel, self.N, self.N))
+        Nby2_percent = self.N * self.percent / 2
+        low = int(self.N / 2 - Nby2_percent)
+        high = int(self.N / 2 + Nby2_percent + 1)
+        center_Img = Img[:, low:high, low:high]
+        cutoff_mask = mask[low:high, low:high]
+        locations = np.where(cutoff_mask.reshape(cutoff_mask.size))
+        center_Img = center_Img.reshape(
+            (center_Img.shape[0], cutoff_mask.size))[:, locations[0]]
+        thresh = self.percent * 0.5
+        data_thresholded, samples_thresholded = \
+            extract_k_space_center_and_locations(
+                data_values=Img,
+                samples_locations=samples,
+                thr=(thresh, thresh),
+                img_shape=(self.N, self.N))
```

Comment:
this has not been done with `is_fft=True`, so it's not testing right, is it?

Reply:
So we either use it when is_fft is True or if rank of input matrix is greater than 2. I can test anyone of these cases, in my opinion (in fact based on current codes), we generally just pass complete matrix, and it proceeds with just a user warning. 

https://github.com/chaithyagr/pysap-mri/blob/d72819910239d4788dcce01ae672b47541e2ebed/mri/reconstructors/utils/extract_sensitivity_maps.py#L130-L136

As you can see, we use is_fft only if we are using FFT mode for gridding which is not default. 

So I feel it is more important to test this case.

Reply:
I see makes sense.